[SDK-519] Updating ABKIDFADelegate documentation with iOS 14 changes. (#1386)
Appboy/braze-docs: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md

- You are attributing app installation to a previously served advertisement
- You are attributing an action within the application to a previously served advertisement

## iOS 14 AppTrackingTransparency

> Beginning with iOS 14 access to the advertising identifier is now on a per app basis. In addition to this delegate, your application will need to prompt for this authorization. For more information you can see this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).

Reviewer: Added some commas/hyphens and slight wording updates

```suggestion
Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to this delegate, your application will need to prompt for this authorization. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview on Privacy](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).
```

> Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to this delegate, your application will need to prompt for this authorization. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview on Privacy](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).

Reviewer 2: `this delegate` is a little confusing and vague

Reviewer 2: also will the iOS 14 guide be a thing once iOS 14 is released (@davidbielik )? we should probably just put the info here

Reviewer: Based off Chava's comment, here's a rewording of the second sentence:

```suggestion
Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. 
Your application will need to prompt the user for authorization to track them as well as implement additional code to handle this. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview on Privacy](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).
```

Author: I'm also thinking of going through and clarify that it's really Xcode 11 vs 12

> Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to implementing Braze's `ABKIDFADelegate` protocol, your application will need to request authorization using Apple's `ATTrackingManager` in the App Tracking Transparency framework. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency). With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA wil Xcode 11 is not possible in iOS 14.

Reviewer 3: > With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA wil Xcode 11 is not possible in iOS 14.

This is really well summarized, going to add this to the upgrade guide, too.

> Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to implementing Braze's `ABKIDFADelegate` protocol, your application will need to request authorization using Apple's `ATTrackingManager` in the App Tracking Transparency framework. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency). With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA wil Xcode 11 is not possible in iOS 14.

Reviewer 3: ```suggestion
Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to implementing Braze's `ABKIDFADelegate` protocol, your application will need to request authorization using Apple's `ATTrackingManager` in the App Tracking Transparency framework. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency). With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA will Xcode 11 is not possible in iOS 14.
```

```objc
#import "IDFADelegate.h"
#import <AdSupport/ASIdentifierManager.h>
#import <AppTrackingTransparency/AppTrackingTransparency.h>

@implementation IDFADelegate

- (NSString *)advertisingIdentifierString {
  return [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
}

- (BOOL)isAdvertisingTrackingEnabledOrATTAuthorized {
  if (@available(iOS 14, *)) {
    return [ATTrackingManager trackingAuthorizationStatus] == ATTrackingManagerAuthorizationStatusAuthorized;
  }
  return [[ASIdentifierManager sharedManager] isAdvertisingTrackingEnabled];
}

> 

Reviewer: nit: remove newline

```swift
import Appboy_iOS_SDK
import AdSupport
import AppTrackingTransparency

class IDFADelegate: NSObject, ABKIDFADelegate {

   func advertisingIdentifierString() -> String {
    return ASIdentifierManager.shared().advertisingIdentifier.uuidString
  }

  func isAdvertisingTrackingEnabledOrATTAuthorized() -> Bool {
    if #available(iOS 14, *) {
      return ATTrackingManager.trackingAuthorizationStatus ==  ATTrackingManager.AuthorizationStatus.authorized
    }
    return ASIdentifierManager.shared().isAdvertisingTrackingEnabled
  }

}

> 

Reviewer: nit: remove newline
